symengine/basic.h

```diff
@@ -220,6 +220,7 @@ bool is_a_sub(const Basic &b);
 RCP<const Basic> expand(const RCP<const Basic> &self);
 umap_short_basic series(const RCP<const Basic> &ex, const RCP<const Symbol> &var, unsigned int prec);
 umap_short_basic series_invfunc(const RCP<const Basic> &ex, const RCP<const Symbol> &var, unsigned int prec);
+void as_numer_denom(const RCP<const Basic> &x, const Ptr<RCP<const Basic>> &numer, const Ptr<RCP<const Basic>> &denom);
```

Comment:
Why not move this into `numer_denom.h`? This algorithm doesn't need anything from the core, so I think it's cleaner if it lives separate from the core, isn't it?